packages/twenty-server/src/engine/core-modules/workspace-trusted-domain/services/workspace-trusted-domain.service.ts

```
@@ -0,0 +1,204 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import crypto from 'crypto';
+
+import { render } from '@react-email/render';
+import { Repository } from 'typeorm';
+import { APP_LOCALES } from 'twenty-shared';
+import { SendTrustDomainValidation } from 'twenty-emails';
+
+import { WorkspaceTrustedDomain as WorkspaceTrustedDomainEntity } from 'src/engine/core-modules/workspace-trusted-domain/workspace-trusted-domain.entity';
+import { WorkspaceTrustedDomain } from 'src/engine/core-modules/workspace-trusted-domain/dtos/trusted-domain.dto';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { User } from 'src/engine/core-modules/user/user.entity';
+import { isWorkDomain } from 'src/utils/is-work-email';
+import { EmailService } from 'src/engine/core-modules/email/email.service';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+import { DomainManagerService } from 'src/engine/core-modules/domain-manager/services/domain-manager.service';
+import { workspaceTrustedDomainValidator } from 'src/engine/core-modules/workspace-trusted-domain/workspace-trusted-domain.validate';
+import {
+  WorkspaceTrustedDomainException,
+  WorkspaceTrustedDomainExceptionCode,
+} from 'src/engine/core-modules/workspace-trusted-domain/workspace-trusted-domain.exception';
+
+@Injectable()
+// eslint-disable-next-line @nx/workspace-inject-workspace-repository
+export class WorkspaceTrustedDomainService {
+  constructor(
+    @InjectRepository(WorkspaceTrustedDomain, 'core')
+    private readonly workspaceTrustedDomainRepository: Repository<WorkspaceTrustedDomainEntity>,
+    private readonly emailService: EmailService,
+    private readonly environmentService: EnvironmentService,
+    private readonly domainManagerService: DomainManagerService,
+  ) {}
+
+  private checkIsVerified(
+    domain: string,
+    inWorkspace: Workspace,
+    fromUser: User,
+  ) {
+    if (!isWorkDomain(domain)) return false;
```

Do we return an explicit error if it's not a work domain or just fail silently?